kratos/python_scripts/assign_vector_variable_process.py

```diff
@@ -67,11 +67,30 @@ def __init__(self, Model, settings ):
 
         self.model_part = Model[settings["model_part_name"].GetString()]
 
-        self.aux_processes = []
-
+        # Get domain size
+        # Note that we check both the current model part and the root one
+        if self.model_part.ProcessInfo.Has(KratosMultiphysics.DOMAIN_SIZE):
+            domain_size = self.model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+            if domain_size not in [2,3]:
+                root_model_part = self.model_part.GetRootModelPart()
+                if root_model_part.ProcessInfo.Has(KratosMultiphysics.DOMAIN_SIZE):
+                    domain_size = root_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+                else:
+                    raise Exception(f"Domain size is found neither in '{self.model_part.Name}' nor in root model part '{self.model_part.GetRootModelPart().Name}' ProcessInfo containers.")
+        else:
+            root_model_part = self.model_part.GetRootModelPart()
```

Comment:
not sure if this is necessary since the ProcessInfo is shared btw parent and SubModelParts

Reply:
Not necessarily. Depends on how you have created the model part (I'm thinking about the case of custom modelers, which is expected to be more common as we move towards the multistage).

Reply:
unless you deliberately change it, it is the same:
https://github.com/KratosMultiphysics/Kratos/blob/fbc8cc2c5d18cef13ca9c5e0357b86a7c7ed5dbb/kratos/sources/model_part.cpp#L2013

Reply:
Imagine this case. You have an mdpa and you parse it. At this point you have the model part and all its submodelparts with a pointer to the same empty `ProcessInfo`. Now you want to assign the `ProcessInfo` from another model part to this model part, so you call the `SetModelProcessInfo`. This sets the provided process info only in the current model part instance, not in its submodelparts. I faced this while developing a custom modeler, so I think it's a good idea to check the root if not found. At the end, it doesn't harm and it's on the safe side.